drivers/cleanup.py

```diff
@@ -3263,15 +3285,17 @@ def gc_force(session, srUuid, force=False, dryRun=False, lockSR=False):
 
 
 def get_state(srUuid):
-    """Return whether GC/coalesce is currently running or not. The information
-    is not guaranteed for any length of time if the call is not protected by
-    locking.
+    """Return whether GC/coalesce is currently running or not. This asks systemd for
+    the state of the templated SMGC service and will return True if it is "activating"
+    or "running" (for completeness, as in practice it will never achieve the latter state)
     """
-    init(srUuid)
-    if lockGCActive.acquireNoblock():
-        lockGCActive.release()
-        return False
-    return True
+    sr_uuid_esc = srUuid.replace("-", "\\x2d")
+    cmd=[ "/usr/bin/systemctl", "is-active", f"SMGC@{sr_uuid_esc}"]
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+    state = result.stdout.decode('utf-8').rstrip()
+    if state == "activating" or state == "running":
+        return True
+    return False
```

Comment:
Looks good, do we need to consider the case where `systemd` is not being used? I'm inclined to say not as I think all consumers of this code are now using `systemd`